lib/srv/sess.go

```diff
@@ -1257,10 +1264,13 @@ func (s *session) removeParty(p *party) error {
 	// "x-teleport-event" channel in the SSH connection.
 	s.emitSessionLeaveEvent(p.ctx)
 
-	// If the leaving party was the last one in the
-	// session, start the lingerAndDie goroutine.
+	// If the leaving party was the last one in the session, start the lingerAndDie
+	// goroutine. Parties that join during the linger duration will cancel the
+	// goroutine to prevent the session from ending with active parties.
 	if len(s.parties) == 0 && !s.isStopped() {
-		go s.lingerAndDie(p)
+		ctx, cancel := context.WithCancel(s.serverCtx)
```

Comment:
`removeParty` is only called under session lock from `party.closeUnderSessionLock`. It's in the comment already, but I've changed the name to `removePartyUnderLock` to make it extra clear to avoid future race conditions.